crates/re_viewport/src/viewport.rs

```diff
@@ -146,19 +146,14 @@ impl<'a, 'b> Viewport<'a, 'b> {
             ctx,
             space_views: &mut blueprint.space_views,
             maximized: &mut blueprint.maximized,
+            has_been_user_edited: &mut blueprint.has_been_user_edited,
         };
 
         ui.scope(|ui| {
             ui.spacing_mut().item_spacing.x = re_ui::ReUi::view_padding();
 
             re_tracing::profile_scope!("tree.ui");
-            let tree_before = tree.clone();
             tree.ui(&mut tab_viewer, ui);
-
-            // Detect if the user has moved a tab or similar:
-            if !blueprint.has_been_user_edited && *tree != tree_before {
-                blueprint.has_been_user_edited = true;
-            }
```

Comment:
This was an essential fix in https://github.com/rerun-io/rerun/pull/3295 for https://github.com/rerun-io/rerun/issues/2880

Reply:
Maybe a better fix is to have two bools - one for deciding wether to auto-add new stuff, and one for deciding wether to do full re-layout

Reply:
let's try that!